man/fi_poll.3.md

```diff
@@ -182,6 +194,23 @@ being closed, otherwise the call will return -FI_EBUSY.
 Waits on a wait set until one or more of its underlying wait objects
 is signaled.
 
+## fi_control
+
+The fi_control call is used to access provider or implementation specific
+details of the wait set. Access to the wait set should be serialized across
+all calls when fi_control is invoked, as it may redirect the implementation
+of wait set operations. The following control commands are usable with a
+wait set.
+
+*FI_GETWAIT (void \*\*)*
+: This command allows the user to retrieve the low-level wait object
+  associated with the wait set. The format of the wait set is specified
+  during wait set creation, through the wait set attributes. The fi_control
+  arg parameter should be an address where a pointer to the returned wait
+  object will be written. This should be an 'int *' for FI_WAIT_FD,
+  or 'struct fi_mutex_cond' for FI_WAIT_MUTEX_COND. Support for FI_GETWAIT
+  is provider specific and may fail if not supported.
```

Comment:
and maybe add here that the call may fail if the wait set is implemented using more than one wait object.

If we need to, we can add a new fi_control operation (e.g. FI_GETWAIT_SET) to get a list of wait items.  This way, FI_GETWAIT doesn't change based on the fid type.